composer/trainer/trainer.py

```diff
@@ -1014,6 +1014,15 @@ def __init__(
         # Logger
         self.logger = Logger(state=self.state, destinations=loggers)
 
+        if save_latest_filename is not None:
+            if any(
+                    isinstance(logger_destination, RemoteUploaderDownloader) and
+                    logger_destination.file_path_format_string != '{remote_file_name}'
+                    for logger_destination in self.logger.destinations):
```

Comment:
nit: maybe move part of this into a separate variable? possibly the list comprehension part? it's a kinda nasty if statement thats a little hard to parse

Reply:
refactored, check and see if looks good (both cases)